### PR TITLE
[FEAT] Waitlist API - BACKEND

### PR DESCRIPTION
**Pull request by EAZYLINK**
# What does the PR do?
This PR implements an API endpoint to receive and store user information submitted through the waitlist signup form on the website

## How should this endpoint be tested?
Send a POST request to /api/v1/waitlist with the request body as follows:
```
{
"email": "string",
"fullName": "string"
}
```
## Checklist of what I did

- [ ] Implemented the endpoint to join a waitlist
- [ ] Validated the request body and returned validation errors if anyone fails
- [ ] Prevented duplicate entry by making the email unique
- [ ] Prevented empty entry for full name by adding non-empty decorator
- [ ] Stored the user entry in the database
- [ ] Implemented a send mail logic to send confirmation mail to registered users
- [ ] Returned a success response for successful operation
- [ ] Added unit test for waitlist service
## Reference Issue
https://github.com/hngprojects/hng_boilerplate_nestjs/issues/106

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -13,6 +13,6 @@ DB_ENTITIES=dist/src/modules/**/entities/**/*.entity{.ts,.js}
 DB_MIGRATIONS=dist/**/migrations/*{.ts,.js}
 DB_TYPE=postgres
 DB_SSL=true
-
+CLIENT_URL=
```
Comment (Cobidex) [inline .env.example:16]: whats this for, a variable with no valu, please clarify or remove ?

**Review by Cobidex — COMMENTED** [review]
please look through to be sure you didnt touch the github actions configuration file

File: .env.example (hunk 1 above)
Comment (EAZYLINK) [reply .env.example:16]: It's used in the waitlist email sent to the user for accessing the website for more information.

**Comment by EAZYLINK** [comment]
> please look through to be sure you didnt touch the github actions configuration file

Github action was not touched, changes made was only on the waitlist module I'm working on.

**Pull request closed without merging**